tests/tests.rs

```diff
@@ -327,6 +328,36 @@ impl ClientTest {
                 .send()
                 .await
                 .unwrap();
+            assert_eq!(rsp.object_size, *size);
```

Comment:
Ah ok, let me update the error handling and tests for known and unknown length streams.

I want to return object size as part of this API because it is a higher level API that supports unknown length streams as well. It is useful for the library user to have this without needing a stat object call.

It is not being added for the lower level S3 API like single part put or complete mp. Planning to move those to a separate module.

Reply:
As the value of truth is at server side, for unknown sized stream, the right way to get size of the stream after uploading is getting it from stat_object. Like in other SDKs, we do not return size in put object. I think the APIs should have same behavior.

Reply:
Here, we just read and uploaded the stream - at the end of the operation we know the size of the uploaded object - we know the right value here as well. We do provide it in golang - https://pkg.go.dev/github.com/minio/minio-go/v7#UploadInfo is the return type of https://pkg.go.dev/github.com/minio/minio-go/v7#Client.PutObject

Reply:
It is also a hassle to do statObject - to do it correctly we need to use If-Match header or Version-Id to get info about the object that was just uploaded. This is much simpler for the purpose of getting the size.